app/components/course/CourseDropDownButton.tsx

```diff
@@ -1,41 +1,82 @@
 import React, { useEffect, useState } from 'react';
+import { CourseCardProps } from "@/app/model/CourseInterface";
+import { useAppContext } from "../../contexts/appContext/AppProvider";
+import { SingleCourse } from "@/app/model/CourseInterface";
 
-const CourseCardDropDown = () => {
-  const [isOpen, setIsOpen] = useState(false);
-  const [dropDownText, setDropDownText] = useState("No Selection");
+const colorMap = {
+  "No Selection": "bg-primary-700 text-white",
+  "In Progress": "bg-orange-500 text-white",
+  "Completed": "bg-green-500 text-white",
+  "Planned": "bg-gray-500 text-white", 
+};
 
-  const handleMouseEnter = () => {
-    setIsOpen(true);
-  };
+const CourseCardDropDown = ({
+  title,
+  courseCode,
+  status = "No Selection"
+}: CourseCardProps) => {
+  const [isOpen, setIsOpen] = useState(false);
+  const { coursesSelected, setCoursesSelected } = useAppContext();
+  status = coursesSelected.find(course => course.courseID === courseCode)?.status || "No Selection";
+  const [dropDownText, setDropDownText] = useState<string>(status);
+  
+  
 
-  const handleMouseLeave = () => {
-    setIsOpen(false);
-  };
+  const updateCourseInContext = (newState: string) => {
+    console.log(coursesSelected);
+    let to_add : SingleCourse = { courseID: courseCode, title: title, status: newState };
+    const updatedCourses = coursesSelected.map(course =>
+      course.courseID === courseCode
+        ? { ...course, status: newState }
+        : course);
+    let selectedIDS = updatedCourses.map(course => course.courseID);
 
-  const handleOption1 = () => {
-    setIsOpen(false);
-    setDropDownText("Planned");
-  };
-  const handleOption2 = () => {
-    setIsOpen(false);
-    setDropDownText("In Progress");
+    if (!selectedIDS.includes(to_add.courseID)) {
+      updatedCourses.push(to_add);
+    }
+    console.log(updatedCourses);
+    setCoursesSelected(updatedCourses);
```

Comment:
This logic should be in the ContextProvider.

You can just move this logic in, pass courseId as argument and update there

Reply:
Fixed. Did it in a slightly different way, by passing in the SingleCourse object that is to be changed.

Reply:
Ok, that's also good. The reason is not to have the update logic in the client (component) file.